pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -724,7 +742,21 @@ func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.Contr
 			return nil, fmt.Errorf("could not update registries config with new changes: %v", err)
 		}
 	}
-	registriesIgn := createNewRegistriesConfigIgnition(registriesTOML)
+	if blockedRegs != nil || allowedRegs != nil {
+		dataURL, err := dataurl.DecodeString(originalPolicyIgn.Contents.Source)
+		if err != nil {
+			return nil, fmt.Errorf("could not decode original policy json: %v", err)
+		}
+		policyJSON, err = updatePolicyJSON(dataURL.Data, blockedRegs, allowedRegs)
+		if err != nil {
+			return nil, fmt.Errorf("could not update policy json with new changes: %v", err)
+		}
+	}
```

Comment:
(It feels a bit off to have `blockedRegs` enforced in two configurations… OTOH it does not hurt anything.)

Reply:
It is needed in registries.conf and policy.json, so going to leave it for now since it is not causing any problems.